### ISSUE TITLE
Imports from one cell are not picked up in the next (interactive window)

### ISSUE THREAD

**Issue by rchiodo** [microsoft/pylance-release#3103]
## Environment data

Name: Pylance
Id: ms-python.vscode-pylance
Description: A performant, feature-rich language server for Python in VS Code
Version: 2022.7.40
Publisher: Microsoft
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=ms-python.vscode-pylance

## Repro Steps

Just run two cells in the interactive window.

First one:
```python
import os
print(os.getenv("FOO"))
```
Second one:
```python
print(os.getenv("FOO"))
```

It's showing an import warning on the second cell.

![image](https://user-images.githubusercontent.com/19672699/181303182-fa6492ff-67c3-4cea-9dc3-b56f8ab8015a.png)

Here's my settings.json:

```json
{
  "extensions.autoUpdate": false,
  "git.enableSmartCommit": true,
  "git.confirmSync": false,
  "python.logLevel": "Trace",
  "python.trace.server.verbosity": "Verbose",
  "notebook-intellisense.logLevel": "Trace",
  "workbench.colorCustomizations": {
    "editorSuggestWidgetStatus.foreground": "#ff0000"
  },
  "editor.suggest.showStatusBar": true,
  "jupyter.logKernelOutputSeparately": true,
  "notebook-intellisense.trace.server.verbosity": "Verbose",
  "python.experiments.enabled": true,
  "jupyter.disableZMQSupport": true,
  "handydandy-notebook.dispatch": {
    "python": ["python", ["-c", "${code}"]],
    "typescript": [
      "C:\\Users\\aku91\\AppData\\Roaming\\npm\\ts-node.cmd",
      ["-e", "(async () => { ${code} } )()"]
    ],
    "javascript": [
      "C:\\PROGRA~1\\nodejs\\node.exe",
      ["-e", "(async () => { ${code} } )()"]
    ],
    "ruby": ["ruby", ["-e", "${code}"]],
    "shellscript": ["bash", ["-c", "${code}"]],
    "bash": ["bash", ["-c", "${code}"]]
  },
  "jupyter.askForKernelRestart": false,
  "[jsonc]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  },
  "handydandy-notebook.defaultLang": "javascript",
  "notebook.lineNumbers": "on",
  "jupyter.interactiveWindowMode": "perFile",
  "python.formatting.provider": "black",
  "editor.suggestSelection": "first",
  "vsintellicode.modify.editor.suggestSelection": "automaticallyOverrodeDefaultValue",
  "problems.compareOrder": "position",
  "editor.accessibilitySupport": "off",
  "[typescript]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  },
  "[typescriptreact]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  },
  "typescript.updateImportsOnFileMove.enabled": "always",
  "typescript.referencesCodeLens.enabled": true,
  "typescript.referencesCodeLens.showOnAllFunctions": true,
  "jupyter.logging.level": "verbose",
  "python.logging.level": "debug",
  "[python]": {
    "editor.defaultFormatter": "ms-python.python",
    "editor.acceptSuggestionOnCommitCharacter": false
  },
  "editor.unicodeHighlight.ambiguousCharacters": false,
  "jupyter.newCellOnRunLast": false,
  "testing.automaticallyOpenPeekView": "failureAnywhere",
  "problems.showCurrentInStatus": true,
  "editor.formatOnSave": true,
  "julia.symbolCacheDownload": true,
  "julia.executablePath": "D:\\Tools\\julia-1.7.1\\bin\\julia.exe",
  "julia.enableTelemetry": true,
  "jupyter.sendSelectionToInteractiveWindow": true,
  "git.repositoryScanMaxDepth": 2,
  "experiments.override.gettingStarted.overrideCategory.ms-python.python.pythonWelcome.when": "true",
  "experiments.override.gettingStarted.overrideCategory.ms-python.python.pythonWelcomeWithDS.when": "workspacePlatform != webworker",
  "experiments.override.gettingStarted.overrideCategory.ms-toolsai.jupyter.jupyterWelcome.when": "workspacePlatform != webworker",
  "[javascript]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  },
  "jupyter.pythonExportMethod": "direct",
  "remote.SSH.remotePlatform": {
    "20.115.44.103": "linux"
  },
  "notebook.output.textLineLimit": 100,
  "remoteHub.logLevel": "info",
  "javascript.updateImportsOnFileMove.enabled": "always",
  "jupyter.runStartupCommands": ["%load_ext autoreload", "%autoreload 2"],
  "githubPullRequests.createOnPublishBranch": "never",
  "workbench.editor.enablePreview": false,
  "notebook.undoRedoPerCell": false,
  "python.tensorBoard.logDirectory": "${workspaceRoot}\\Foo",
  "editor.tabCompletion": "onlySnippets",
  "terminal.integrated.shellIntegration.enabled": true,
  "python.pylanceLspNotebooksEnabled": true,
  "python.server.traces": "verbose",
  "python.traces.server": "verbose",
  "terminal.integrated.windowsEnableConpty": false,
  "terminal.integrated.showExitAlert": false,
  "terminal.integrated.gpuAcceleration": "off",
  "terminal.integrated.environmentChangesRelaunch": false,
  "jupyter.enablePreciseKernelCompletions": true,
  "jupyter.pythonCompletionTriggerCharacters": ".%\"'",
  "jupyter.enableExtendedKernelCompletions": true,
  "jupyter.forceIPyKernelDebugger": false,
  "window.zoomLevel": -1,
  "editor.renderWhitespace": "all",
  "editor.inlineSuggest.enabled": true,
  "github.copilot.enable": {
    "typescript": true,
    "yaml": false,
    "plaintext": false,
    "markdown": false
  },
  "testing.alwaysRevealTestOnStateChange": true,
  "jupyter.widgetScriptSources": ["jsdelivr.com", "unpkg.com"],
  "jupyter.showOnlyOneTypeOfKernel": false,
  "workbench.colorTheme": "Default Light+",
  "diffEditor.ignoreTrimWhitespace": false,
  "jupyter.diagnostics.reservedPythonNames.exclude": [
    "**/site-packages/**",
    "**/lib/python/**",
    "**/lib64/python/**",
    "d:\\source\\debugpy\\src\\debugpy\\adapter\\__main__.py",
    "d:\\source\\debugpy\\src\\debugpy\\launcher\\__main__.py"
  ],
  "git.mergeEditor": false,
  "githubPullRequests.pullBranch": "never"
}
```

## Logs

```
XXX
```

**Comment by sbatten** [comment]
Not sure if this is the same issue or a separate issue, but using an interactive notebook, function definitions are not carried across cells.

![image](https://user-images.githubusercontent.com/6561887/181350651-1ad39546-3c82-4bd1-8af2-53752f56fe4c.png)


first cell defines `foo()`
second calls `foo()`

execution is correct

**Comment by debonte** [comment]
This was fixed in [2022.8.10](https://github.com/microsoft/pylance-release/releases/tag/2022.8.10).
